Add registered_provider_plugins to Vmdb::Plugins

durandom opened the pull request:
this allows for querying which providers are registered via plugins.
it also provides a helper method to query for provider names of all registered providers

extracted from https://github.com/ManageIQ/manageiq/pull/13942

@miq-bot add_labels pluggable providers, enhancement
@miq-bot assign @bdunne

bdunne commented on lib/vmdb/plugins.rb:
```diff
@@ -4,29 +4,47 @@ class Plugins
-    def register_automate_domains(engine)
-      Dir.glob(engine.root.join("content", "automate", "*")).each do |domain_directory|
-        @registered_automate_domains << AutomateDomain.new(domain_directory)
+    def provider_plugin_names
```
We usually implement something like this as:
```ruby
def provider_plugin_names(refresh = false)
  @provider_plugin_names = nil if refresh
  @provider_plugin_names ||= .....
end
```

bdunne commented on lib/vmdb/plugins.rb:
```diff
@@ -4,29 +4,47 @@ class Plugins
+    def register_provider_plugin(engine)
+      if engine.class.name.start_with?("ManageIQ::Providers::")
+        @registered_provider_plugins << engine
+        @provider_plugins_names = nil
```
It feels strange to wipe this cache in this method.  Based on the comment above, you could instead call `provider_plugin_names(true)`

bdunne commented on lib/vmdb/plugins.rb:
```diff
@@ -4,29 +4,47 @@ class Plugins
-        @registered_automate_domains << AutomateDomain.new(domain_directory)
+    def provider_plugin_names
+      @provider_plugins_names ||= registered_provider_plugins.collect do |plugin|
+        ManageIQ::Providers::Inflector.provider_name(plugin).downcase.to_sym
```
Should this be `underscore` rather than `downcase` to deal with things like `AnsibleTower` in the future?

durandom commented:
@bdunne addressed your feedback - thanks for catching the `underscore` issue
also changed to storing the provider plugins in a hash to circumvent the cache invalidation issue
